File: observer/typescript/product-update-message.ts
// Bildirilecek olan nesnelere paremetre olarak giden mesaj.
// UML diyagramı ile ilgisi yoktur.
export class ProductUpdateMessage {
  public productName: string;
  public message: string;

  public toString(): string {
    return `${this.productName} updated. Message: ${this.message}`;
  }
}

File: observer/go/product-update-message.go
package main

import "fmt"

// ProductUpdateMessage Bildirilecek olan nesnelere paremetre olarak giden mesaj.
// UML diyagramı ile ilgisi yoktur.
type ProductUpdateMessage struct {
	ProductName string
	Message     string
}

func NewProductUpdateMessage() ProductUpdateMessage {
	return ProductUpdateMessage{}
}

func (m ProductUpdateMessage) ToString() string {
	return fmt.Sprintf("%s updated. Message: %s.", m.ProductName, m.Message)
}
